mlflow/tracking/_default_experiment/abstract_context.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABCMeta, abstractmethod
+
+
+class DefaultExperimentProvider(object):
+    """
+    Abstract base class for context provider objects specifying experiment_id at run-creation time
+    (e.g. experiment_id created for the job context for which the experiment is created).
+
+    When an experiment is created via the fluent ``mlflow.start_run`` method,
+    MLflow iterates through all registered DefaultExperimentProvider.
+    For each context provider where ``in_context`` returns ``True``,
+    MLflow calls the ``get_experiment_id`` method on the context provider
+    to compute experiment_id for the experiment.
```

Comment:
`_get_experiment_id()` is also used in other places, e.g. MLflow Projects. Can we discuss how it's used by `_get_experiment_id()` and cite ``mlflow.start_run()`` as just one example of where `_get_experiment_id()` is used?

Reply:
Can we also explain that this is used to resolve a default experiment when the user has not explicitly set one themselves?

Reply:
Can we lengthen the line width here?